p2p/src/peer.rs

```diff
@@ -250,13 +235,14 @@ impl Peer {
 			total_difficulty,
 			height,
 		};
-		connection!(self).send(ping_msg, msg::Type::Ping)
+		self.connection.lock().send(ping_msg, msg::Type::Ping)
```

Comment:
I wonder should we introduce `send` method on Peer? This construction is verbose and reveal a lot of internal structure of the object, which is against law of Demeter, which is not a law of course, but a nice principle to follow https://hackernoon.com/object-oriented-tricks-2-law-of-demeter-4ecc9becad85

Reply:
:+1: doing this now.